File: A2_q1a.cxx
#include<iostream>
using namespace std;

int distance(int x1, int x2){
	if((x2-x1)>=0) return (x2-x1);
	else return (x1-x2);
}

int main(){
int n;
cout<<"To find the average distance between 2 points on a straight line made up of N discrete points having eqaul interval of 1 unit."<<endl<<endl;
cout<<"Enter the number(natural number) of points: ";
cin>>n;

float sum= 0, m=0;
if(n>0){
	for(int i=0; i<n; i++ ){
		for(int j=0; j<n; j++){
			sum=sum+distance(i,j);
			m++;
		}
	}
float avg;
avg = sum/m;

cout<<"The average distance between 2 points is "<<avg<<" unit."<<endl;
}
else cout<<"It is not possible to create a chain with the entered number of points. ";

return 0;
}

File: A2_q1a.py
sum=0
m=0
print("To find the average distance between 2 points on a straight line made up of N discrete points having eqaul interval of 1 unit.")
y = input('Enter the number of points in that chain: ')
try:
	n=int(y)
	if n>=0:
		for i in range(0,n):
			for j in range(0, n):
				if(i>=j):
					sum+=i-j
					m+=1
				else:
					sum+=j-i
					m+=1
		print('The average distance between 2 points is: ', sum/m)		
	else:
		print('The input value is not a whole number')
except ValueError:
	print("It is not possible to create a chain with the entered value. ")